### PR TITLE
Stored Procedure Blog Post

### PR DESCRIPTION
**Pull request by bmrsny**
My blog post for 06-17-16

### CODE REVIEW

File: _posts/2016-06-17-stored-procedures.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+banner_image: sql.jpg
+---
+
+
+
+I was doing some routine code review this past week when I came across something I wasn't familiar with. A very cryptic looking chunk of SQL. It turns out, the code I was looking at was a Stored Procedure.
+
+<img src="{{ site.url }}/assets/images/procedure.jpg">
+
+What is a Stored Procedure? A Stored Procedure is a set SQL statements with an assigned name that's stored in the database in compiled form so that it can be shared by a number of programs. This is a concept should be familiar, if you have been developing for any bit of time, method definition and execution. Sounds like a great way to DRY up some SQL queries or aggregations.
```
Comment (doomspork) [inline _posts/2016-06-17-stored-procedures.md]: "is a set _of_ SQL"

Comment (rshea303) [inline _posts/2016-06-17-stored-procedures.md]: 3rd sentence....  'This concept should be familiar' or 'This is a concept that should be familiar...'

File: _posts/2016-06-17-stored-procedures.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  INSERT INTO person (person_id, first_name, last_name)
+  VALUES
+  (1, 'Harry', 'Potter'),
+  (2, 'Tom', 'Riddle'),
+  (3, 'Hermoine', 'Granger'),
+  (4, 'Albus', 'Dumbledore'),
+  (5, 'Ron', 'Weasley');
+{% endhighlight %}
+
+So we created a db named my_db, added a person table, and filled that person table with 5 Harry Potter Pals.
```
Comment (rshea303) [inline _posts/2016-06-17-stored-procedures.md]: I don't know if Tom Riddle would be considered a pal.

File: _posts/2016-06-17-stored-procedures.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  END $$
+  DELIMITER;
+
+  CALL get_harry('Harry', @harry);
+  SELECT @harry;
+  // returns @harry set to 'Harry'
+{% endhighlight %}
+
+In the above statement, `INTO`, sets the second argument passed in to the first_name value found.
+Since the second value passed in is `@Harry`, this variable is set inside the SQL statement. Now we can get the value stored in `@harry` from running the procedure.
```
Comment (rshea303) [inline _posts/2016-06-17-stored-procedures.md]: Is it `@Harry` or `@harry` or just `Harry` for the second value being passed in?

File: _posts/2016-06-17-stored-procedures.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+
+`SELECT @harry;   // returns @harry set to 'Harry'`
+
+Now that we have an understanding of IN and OUT, I will let you try and work through and INOUT example.
```
Comment (rshea303) [inline _posts/2016-06-17-stored-procedures.md]: 'and' should be 'an' at end of sentence.    work through an INOUT example

File: _posts/2016-06-17-stored-procedures.md (hunk 2 above)
Comment (bmrsny) [inline _posts/2016-06-17-stored-procedures.md]: I decided to go with Snape.

**Pull request merged**